### ISSUE TITLE
 Mismatched Input

### ISSUE THREAD

**Issue by xiaodiannao** [JesusFreke/smali#725]
hi everyone, thanks for this awesome project. I'm trying to first baksmali the boot-framework.oat file using `java -jar baksmali-2.3.2.jar deodex x86/boot-framework.oat -o out` which works fine. But now I'm trying to assemble the smali files to a dex file using `java -jar smali-2.3.2.jar assemble out/` which results in no output and prints out a lot of errors in the form of: 

```
out/android/accessibilityservice/AccessibilityService$2.smali[177,63] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$2.smali[187,180] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$Callbacks.smali[45,85] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController$1.smali[11,143] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController$OnMagnificationChangedListener.smali[18,147] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController$1.smali[11,146] mismatched input ')' expecting EQUAL
out/android/accessibilityservice/AccessibilityService$MagnificationController$1.smali[12,0] mismatched input '.end annotation' expecting EQUAL
out/android/accessibilityservice/AccessibilityService$MagnificationController$1.smali[35,233] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$IAccessibilityServiceClientWrapper.smali[337,154] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController$1.smali[75,270] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController.smali[189,75] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController.smali[293,328] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$MagnificationController.smali[313,270] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService$IAccessibilityServiceClientWrapper.smali[869,76] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService.smali[178,112] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService.smali[187,141] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService.smali[337,77] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/AccessibilityService.smali[352,172] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/GestureDescription$StrokeDescription.smali[44,57] mismatched input 'J' expecting CLOSE_PAREN
out/android/accessibilityservice/GestureDescription$StrokeDescription.smali[61,135] mismatched input 'J' expecting CLOSE_PAREN
out/android/accessibilityservice/GestureDescription$StrokeDescription.smali[67,57] mismatched input 'J' expecting CLOSE_PAREN
out/android/accessibilityservice/GestureDescription$StrokeDescription.smali[301,53] mismatched input 'J' expecting CLOSE_PAREN
out/android/accessibilityservice/GestureDescription$StrokeDescription.smali[326,135] mismatched input 'J' expecting CLOSE_PAREN
out/android/accessibilityservice/IAccessibilityServiceClient$Stub$Proxy.smali[653,76] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/IAccessibilityServiceClient$Stub.smali[337,154] mismatched input 'F' expecting CLOSE_PAREN
out/android/accessibilityservice/IAccessibilityServiceClient.smali[98,85] mismatched input 'F' expecting CLOSE_PAREN
out/android/accounts/AccountManager$13.smali[149,165] mismatched input 'I' expecting CLOSE_PAREN
out/android/accounts/AccountManager$14.smali[95,187] mismatched input 'Z' expecting CLOSE_PAREN
out/android/accounts/AccountManager$7.smali[139,167] mismatched input 'Z' expecting CLOSE_PAREN
out/android/accounts/AccountManager$8.smali[90,167] mismatched input 'Z' expecting CLOSE_PAREN
out/android/accounts/AuthenticatorDescription.smali[168,70] mismatched input 'I' expecting CLOSE_PAREN
out/android/accounts/AuthenticatorDescription.smali[194,124] mismatched input 'I' expecting CLOSE_PAREN
out/android/accounts/AuthenticatorDescription.smali[200,70] mismatched input 'I' expecting CLOSE_PAREN
```

Does anybody know what the problem here is?
Thanks a lot and best regards

**Comment by sinsongdev** [comment]
Same problem here.

**Comment by JesusFreke** [comment]
Thanks for the report. Fixed by ae2efe146b8c571821365ec7d43e6db070a678d9, and I'm making a new 2.3.3 release for the fix
